plugin/vim/quick-lint-js.vim/lua/lspconfig/quick_lint_js.lua

```diff
@@ -0,0 +1,30 @@
+local configs = require("lspconfig/configs")
+local util = require("lspconfig/util")
+
+configs.quick_lint_js = {
+    default_config = {
+        cmd = {"quick-lint-js", "--lsp-server"},
+        filetypes = {"javascript"},
+        root_dir = function(fname)
+            return util.find_node_modules_ancestor(fname) or
+                       util.find_git_ancestor(fname) or
+                       util.path.dirname(fname)
+        end
+    },
+    docs = {
+        description = [[
+https://quick-lint-js.com/
+quick-lint-js finds bugs in JavaScript programs.
+[200× faster than ESLint](https://quick-lint-js.com/benchmarks/),
+quick-lint-js gives you instant feedback as you code.
+Find bugs in your JavaScript before your finger leaves the keyboard.
+Lint any JavaScript file with no configuration.
+```sh
+npm install [-g] quick-lint-js
```

Comment:
What code causes this plugin to use the quick-lint-js installed by `npm install quick-lint-js`? I see `util.find_node_modules_ancestor(fname)`, but how does that turn into running `node_modules/.bin/quick-lint-js`?

Reply:
That's the  ``cmd = {"quick-lint-js", "--lsp-server"},`` which expects **quick-lint-js** to be in ``$PATH``. It's possible to make it look somewhere and replace it instead, do you have a helper function to find its path?

Reply:
> That's the `cmd = {"quick-lint-js", "--lsp-server"},` which expects **quick-lint-js** to be in `$PATH`.

Right. So why do the instructions recommend running `npm install quick-lint-js` if that won't put `quick-lint-js` in `$PATH`?

> It's possible to make it look somewhere and replace it instead, do you have a helper function to find its path?

I think it'd be `node_modules/.bin/quick-lint-js`. However, this might be a security risk.

What does ESLint's integration do?

Reply:
That's the default configuration. You either put it in ``$PATH`` and use it like this:
```lua
require("lspconfig").quick_lint_js.setup{}
```
or you overwrite the default config in your vimrc/init.lua like this:
```lua
require("lspconfig").quick_lint_js.setup{ cmd = "/somewhere/node_modules/.bin/quick-lint-js" }
```
There are some plugins like [lspinstall](https://github.com/kabouzeid/nvim-lspinstall) or the one I was making [lspconfigplus](https://github.com/tiagovla/lspconfigplus.nvim) that manages installation and update of lsp servers. They pretty much install it somewhere and overwrite that ``cmd`` key to point to the executable, seamlessly.

ESLint is not listed [there](https://github.com/neovim/nvim-lspconfig/tree/master/lua/lspconfig). All of them consider the executable to be in ``$PATH``.